compaction.js

```diff
@@ -134,7 +134,8 @@ function Compaction(log, onDone) {
   const progress = Obv() // for the unshifted offset
   let startOffset = 0
   let version = 0
-  let holesFound = true // assume true
+  let holesFound = 0
+  let hasHoles = true // assume true
```

Comment:
I would probably remove this boolean variable in this function and just use holesFound